src/main/java/com/moodybluez/enterprise/dao/IEntryDAO.java

```diff
@@ -0,0 +1,19 @@
+package com.moodybluez.enterprise.dao;
+
+import com.moodybluez.enterprise.dto.Entry;
+
+import java.util.List;
+import java.util.Map;
+
+public interface IEntryDAO {
+
+    boolean saveEntry(Entry entry);
+
+    Entry fetchByWeekDay(int weekDayID);
+
+    List<Entry> fetchByMood(int moodID);
+
+    Entry fetchByDate(String date);
+
+    Map<Integer, Entry> fetchAll();
```

Comment:
Added Persistence level methods (fully implemented with logic in actual classes) to this layer, moved from service